malaria-ios/Managers/RapidFireAchievementManager.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+
+/// Class that handles all the achievements related to the Rapid Fire Game.
+
+class RapidFireAchievementManager: NSObject {
+  static let sharedInstance = RapidFireAchievementManager()
+  
+  private var game: RapidFireGame?
+  private let achievementManager = AchievementManager.sharedInstance
+  
+  override init() {
+    super.init()
+    
+    defineAchievements()
```

Comment:
To force developers to do this right. You can define a protocol or a abstract class where your achivementsManager have to define "defineAchivements" you might can refactor and provide to the NSNotificationEvents argument to the super class and force the classes to have a "checkAchivement" argument. Is this clear to you? TL ; DR; Abstract class that receives the NSNotificationEvent with defineAchivement and checkAchivement as abstract methods.